sycl/include/CL/sycl/pipes.hpp

```diff
@@ -0,0 +1,89 @@
+//==---------------- pipes.hpp - SYCL context -----------*- C++ -*----------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// ===--------------------------------------------------------------------=== //
+
+#pragma once
+
+#include <CL/__spirv/spirv_ops.hpp>
+#include <CL/__spirv/spirv_types.hpp>
+#include <CL/sycl/stl.hpp>
+
+namespace cl {
+namespace sycl {
+
+template <class name, class dataT, int32_t min_capacity = 0> class pipe {
+public:
+  // Non-blocking pipes
+  // Reading from pipe is lowered to SPIR-V instruction OpReadPipe via SPIR-V
+  // friendly LLVM IR.
+  static dataT read(bool &Success) {
+#ifdef __SYCL_DEVICE_ONLY__
+    RPipeTy<dataT> RPipe =
+      __spirv_CreatePipeFromPipeStorage_read<dataT>(m_Storage);
+    dataT TempData;
+    Success = static_cast<bool>(
+        __spirv_ReadPipe(RPipe, &TempData, m_Size, m_Alignment));
+    return TempData;
+#else
+    assert(!"Pipes are not supported on a host device!");
+#endif // __SYCL_DEVICE_ONLY__
+  }
+
+  // Writing to pipe is lowered to SPIR-V instruction OpWritePipe via SPIR-V
+  // friendly LLVM IR.
+  static void write(dataT Data, bool &Success) {
```

Comment:
`static bool write(const dataT &Data)` instead?

Reply:
Thanks!

Reply:
Well, actually I can't do this change right now due to some documentation reasons.

Reply:
The const reference is good.  The status return code move to returned bool makes the signatures of read and write differ more than I would like.  Read returns the data type in both cases.  How strongly do you think that write should return bool success status, while read returns the data and takes a status ref arg?